core/src/models/Source.ts

```diff
@@ -141,6 +141,8 @@ export class Source extends CommonModel<Source> {
   @BelongsTo(() => GrouparooModel)
   model: GrouparooModel;
 
+  uniqueIdentifier = ["appId", "name"];
```

Comment:
This differs from the previous functionality with the addition of `appId`, let me know if it may be a problem.

Reply:
Does this mean there can't be two sources with the same app? That's not right?
Or am I misreading the effect of this?

Reply:
I think right now the code is ANDing together everything from this list, right?

Reply:
Yeah, this is effectively just a composite unique index.